pkg/formats/sniffer.go

```diff
@@ -46,6 +46,49 @@ func (fs *Sniffer) SniffReader(f io.ReadSeeker) (Format, error) {
 			fmt.Printf("WARNING: could not seek to beginning of file: %v", err)
 		}
 	}()
+
+	type SpecVersionStruct struct {
+		BomFormat       string `json:"bomFormat"`
+		CDXSpecVersion  string `json:"specVersion"`
+		SPDXSpecVersion string `json:"spdxVersion"`
+	}
+
+	decoder := json.NewDecoder(f)
+
+	var specversionjson SpecVersionStruct
+	err := decoder.Decode(&specversionjson)
+	if err == nil {
+		if specversionjson.BomFormat == "CycloneDX" {
+			if specversionjson.CDXSpecVersion == "1.3" {
+				return CDX13JSON, nil
+			} else if specversionjson.CDXSpecVersion == "1.4" {
+				return CDX14JSON, nil
+			} else if specversionjson.CDXSpecVersion == "1.5" {
+				return CDX15JSON, nil
+			} else {
+				// JSON + BomFormat CycloneDX but specVersion not 1.3, 1.4, or 1.5
+				return "", fmt.Errorf("unknown SBOM format")
```

Comment:
If #102 merges before this one, you will be able to replace this with:
```suggestion
   f, err := cdxformats.ParseVersion(specversionjson.CDXSpecVersion)
   if err != nil {
     return "", fmt.Erorf("parsing cyclonedx format: %w", err)
   }
   return f, nil
```

(totally fine to this later)

Reply:
Sure, as soon as #102 merges I'll make that change, thanks!